src/components/Header/web3/NetworkModal/index.tsx

```diff
@@ -33,40 +59,188 @@ export default function NetworkModal({
+    } else {
+      if (favoriteChains.some(fChainId => fChainId === chainId)) {
+        saveFavoriteChains([...favoriteChains.filter(fChainId => fChainId !== chainId)])
+      }
+    }
+  }
+
+  const saveFavoriteChains = (chains: string[]) => {
+    const uniqueArray = Array.from(new Set(chains))
+    requestSaveProfile({ data: { favouriteChainIds: uniqueArray } })
+    setFavoriteChains(uniqueArray)
+  }
+
+  const renderNetworkButton = (networkInfo: NetworkInfo, isAdding: boolean) => {
+    return (
+      <DraggableNetworkButton
+        key={networkInfo.chainId}
+        droppableRefs={droppableRefs}
+        dragConstraints={wrapperRef}
+        networkInfo={networkInfo}
+        activeChainIds={activeChainIds}
+        isSelected={selectedId === networkInfo.chainId}
+        disabledMsg={disabledMsg}
+        onDrop={(dropId: string) => {
+          handleDrop(networkInfo.chainId.toString(), dropId)
+        }}
+        customToggleModal={customToggleModal}
+        customOnSelectNetwork={customOnSelectNetwork}
+        onChangedNetwork={toggleNetworkModal}
+        // Mobile only props
+        isAddButton={isAdding}
+        isEdittingMobile={isEdittingMobile}
+        onFavoriteClick={() => handleFavoriteChangeMobile(networkInfo.chainId.toString(), isAdding)}
+      />
+    )
+  }
+
+  useEffect(() => {
+    setFavoriteChains(userInfo?.data?.favouriteChainIds || [])
+  }, [userInfo])
+
   return (
     <Modal
       isOpen={isOpen !== undefined ? isOpen : networkModalOpen}
       onDismiss={toggleNetworkModal}
       maxWidth={624}
       zindex={Z_INDEXS.MODAL}
+      height="500px"
     >
-      <Wrapper>
-        <Flex alignItems="center" justifyContent="space-between">
+      <Wrapper ref={wrapperRef}>
+        <RowBetween>
           <Text fontWeight="500" fontSize={20}>
             {isWrongNetwork ? <Trans>Wrong Chain</Trans> : <Trans>Select a Chain</Trans>}
           </Text>
-
-          <Flex sx={{ cursor: 'pointer' }} role="button" onClick={toggleNetworkModal}>
+          <ButtonAction onClick={toggleNetworkModal}>
             <X />
-          </Flex>
-        </Flex>
-        {isWrongNetwork && (
-          <TYPE.main fontSize={16} marginTop={14}>
-            <Trans>Please connect to the appropriate chain.</Trans>
-          </TYPE.main>
-        )}
-        <Networks
-          onChangedNetwork={toggleNetworkModal}
-          selectedId={selectedId}
-          activeChainIds={activeChainIds}
-          customOnSelectNetwork={customOnSelectNetwork}
-          customToggleModal={customToggleModal}
-          disabledMsg={disabledMsg}
-        />
+          </ButtonAction>
+        </RowBetween>
+
+        <Column marginTop="16px" gap="8px">
+          <Row gap="12px">
+            <Text fontSize="10px" lineHeight="24px" color={theme.subText} flexShrink={0}>
+              <Trans>Favorite Chain(s)</Trans>
+            </Text>
+            <hr style={{ border: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```

Comment:
`border: '0 0 1px 0'` this is not working. Can you test again

Reply:
```suggestion
            <hr style={{ borderWidth: '0 0 1px 0', borderColor: theme.border, width: '100%' }} />
```

Reply:
yeah right, but it showed perfectly on UI :)))